toolchain/check/handle_file.cpp

```diff
@@ -0,0 +1,21 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/context.h"
+
+namespace Carbon::Check {
+
+auto HandleFileStart(Context& context, Parse::Node parse_node) -> bool {
+  // Push the file as a sentinel so that it's always safe to peek at the
+  // enclosing node on the node stack to determine what context we're in.
```

Comment:
I worry about how `Peek` takes arbitrary values, but you mainly use `Peek(1)`, and so you add a sentinel value for `Peek(1)`. But what if someone starts calling `Peek(2)`, should another sentinel value get added?

Reply:
We should only need a single sentinel. Think of this like null-termination: if we know that level N is not `FileStart`, then it's safe to peek at level N+1. (And if you don't know what level N is, you shouldn't be looking at level N+1, because it wouldn't in general be correct to skip past level N without knowing what it is.)

Reply:
I'm not using `Peek(n)` any more. I think the sentinel still makes sense though, to ensure that our `Peek()` calls are safe. Let me know if you'd prefer that I remove it again; I don't need it in this PR any more.

Reply:
My leaning would be to revert it as unused, since we may not need it. Note for null termination we typically use `StringRef` 😉 .

But it may be that directly checking `N+1` would be incrementally faster than bounds-checking, and if needed, maybe it makes sense on that end. My leaning would still be wait until it's helpful, I'm generally hesitant about the `Peek(N)` model, if it's the best/flexible fit.

Reply:
The sentinel could also be useful for our current uses of `Peek`, which can fail if they happen at the top level. But I've looked through the things that we can handle as a top level declaration, and all of them push something before they do any `Peek`s, so I think all our current `Peek`s are fine.

Removed this push / pop, at least for now. Should be easy to add it back if we find we have a need later.